### PR TITLE
nfs_ganesha: variable validation

### PR DESCRIPTION
**Pull request by Sheersha-jain**
Checking if the given variables are defined or not.

Signed-off-by: Sheersha Jain <shjain@localhost.localdomain>

### CODE REVIEW

File: roles/gluster_hci/tasks/hci_volumes.yml [hunk 1]
```diff
@@ -1,5 +1,21 @@
 ---
 # Create volumes for gluster_hci
+
+- name: check if gluster_features_hci_cluster is set
+  fail:
+    msg: "gluster_features_hci_cluster is undefined"
```
Comment (sac) [inline roles/gluster_hci/tasks/hci_volumes.yml]: @Sheersha-jain can you change the message to:

Variable gluster_features_hci_cluster has to be defined.

File: roles/gluster_hci/tasks/hci_volumes.yml [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+  when: gluster_features_hci_cluster is not defined
+
+- name: check if gluster_features_hci_volume_options is set
+  fail:
+    msg: "gluster_features_hci_volume_options is undefined"
```
Comment (sac) [inline roles/gluster_hci/tasks/hci_volumes.yml]: This variable check you can remove. Because these are set in defaults/main.yml

File: roles/gluster_hci/tasks/hci_volumes.yml [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+  when: gluster_features_hci_volume_options is not defined
+
+- name: check if gluster_features_hci_volumes is set
+  fail:
+    msg: "gluster_features_hci_volumes is undefined"
```
Comment (sac) [inline roles/gluster_hci/tasks/hci_volumes.yml]: Can you change the message to:

Variable gluster_features_hci_volumes has to be defined.

File: roles/nfs_ganesha/tasks/enable_ganesha.yml [hunk 4]
```diff
@@ -27,6 +27,11 @@
   debug: msg={{ result.stderr }} verbosity=0
   when: result.stderr != ''
 
+- name: check if gluster_features_ganesha_volume is set
+  fail:
+    msg: "gluster_features_ganesha_volume is undefined"
```
Comment (sac) [inline roles/nfs_ganesha/tasks/enable_ganesha.yml]: Please change the message as explained above.

File: roles/nfs_ganesha/tasks/main.yml [hunk 5]
```diff
@@ -15,6 +15,11 @@
      ganesha_ha_base_dir: '/var/run/gluster/shared_storage/nfs-ganesha'
   run_once: True
 
+- name: check if gluster_features_ganesha_clusternodes is set
+  fail:
+    msg: "gluster_features_ganesha_clusternodes is undefined"
```
Comment (sac) [inline roles/nfs_ganesha/tasks/main.yml]: Please change the message as explained above.

File: roles/nfs_ganesha/tasks/main.yml [hunk 6]
```diff
@@ -23,6 +28,11 @@
   tags:
     - prerequisites
 
+- name: check if gluster_features_ganesha_masternode is set
+  fail:
+    msg: "gluster_features_ganesha_masternode is undefined"
```
Comment (sac) [inline roles/nfs_ganesha/tasks/main.yml]: Please change the message as explained above.

File: roles/nfs_ganesha/tasks/main.yml [hunk 7]
```diff
@@ -33,6 +43,11 @@
      - secret.pem
   delegate_to: "{{ gluster_features_ganesha_masternode }}"
 
+- name: check if gluster_features_ganesha_newnodes_vip is set
+  fail:
+    msg: "gluster_features_ganesha_newnodes_vip is undefined"
```
Comment (sac) [inline roles/nfs_ganesha/tasks/main.yml]: Please change the message as explained above.

**Comment by sankarshanmukhopadhyay** [comment]
> Checking if the given variables are defined or not.
> 
> Signed-off-by: Sheersha Jain [shjain@localhost.localdomain](https://mail.google.com/mail/?view=cm&fs=1&tf=1&to=shjain@localhost.localdomain)

The sign-off needs to be reconfigured I'd guess.

**Comment by Sheersha-jain** [comment]
@sankarshanmukhopadhyay  @sac I have updated the changes.

File: roles/nfs_ganesha/tasks/main.yml [hunk 8]
```diff
@@ -23,16 +28,26 @@
...
+  fail:
+    msg: "Variable gluster_features_ganesha_masternode has to be defined"
+  when: gluster_features_ganesha_masternode is not defined
+
 # The base directory has to be one of the directories unique in /tmp/
 # Generated in the main.yml of the task
 - name: Copy the ssh keys to the local machine
   fetch: src=/var/lib/glusterd/nfs/{{item}}
-         dest="{{ ganesha_tmp_dir }}"
+         dest= "{{ ganesha_tmp_dir }}"
```
Comment (sac) [inline roles/nfs_ganesha/tasks/main.yml]: @Sheersha-jain the patch looks but. But why did you add a space after `=' here? It will not run, it errors out.

**Comment by Sheersha-jain** [comment]
@sac updated the space fix, please check.
